tests/DependencyInjection/Compiler/MenuBuilderPassTest.php

```diff
@@ -40,47 +40,43 @@ public function testNoopWithoutProvider()
         $this->pass->process($this->containerBuilder->reveal());
     }
 
-    /**
-     * @expectedException \InvalidArgumentException
-     * @expectedExceptionMessage Abstract services cannot be registered as menu builders but "id" is.
-     */
     public function testFailsWhenServiceIsAbstract()
     {
+        $this->expectException(\InvalidArgumentException::class);
+        $this->expectExceptionMessage('Abstract services cannot be registered as menu builders but "id" is.');
+
         $this->builderDefinition->isAbstract()->willReturn(true);
         $this->containerBuilder->findTaggedServiceIds('knp_menu.menu_builder')->willReturn(['id' => [['alias' => 'foo']]]);
 
         $this->pass->process($this->containerBuilder->reveal());
     }
 
-    /**
-     * @expectedException \InvalidArgumentException
-     * @expectedExceptionMessage Menu builder services must be public but "id" is a private service.
-     */
     public function testFailsWhenServiceIsPrivate()
     {
+        $this->expectException(\InvalidArgumentException::class);
+        $this->expectExceptionMessage('Menu builder services must be public but "id" is a private service.');
```

Comment:
Migrating from PHPUnit annotations to PHPUnit methods for exception expectations should be done separately IMO.

Thus, the main benefit of these methods is that they can set the expectation just before the place where we expect it (so that an exception during the setup of the test does not make it pass). So these calls should not be placed at the top of the test, but just before the call to the tested code (the call to `->process()` here).

Reply:
btw, this should be done in the 2.x branch to make it simpler to merge branches together if needed (the new methods are available since PHPUnit 5, which can be used on PHP 7.1)